### ISSUE TITLE
Statistics card unknown error

### ISSUE THREAD

**Issue by TimoJJ** [home-assistant/frontend#15632]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

After upgrading to 2023.3.0b2, and all other versions after that, Statistic Cards stopped working and show just red unknow error text. Same happens also if I try add a new statistics card. 

### Describe the behavior you expected

Work like they did with 2023.2

### Steps to reproduce the issue

1. use or add Statistics card

![image](https://user-images.githubusercontent.com/63199895/221769244-3cc7d4b9-4d08-4763-8cde-96bef4ee6e99.png)

### What version of Home Assistant Core has the issue?

Home Assistant 2023.3.0b5

### What was the last working version of Home Assistant Core?

2023.2

### In which browser are you experiencing the issue with?

MS Edge, Chrome

### Which operating system are you using to run this browser?

Win 10

### Javascript errors shown in your browser console/inspector

```txt
ha-logbook.ts:471 
        
       card statistic Error: Statistic type must be specified
    at p.value (c29b7670.js:1:11659)
    at a (ha-logbook.ts:471:1)
    at ha-logbook.ts:471:1
    at u (ha-logbook.ts:471:1)
    at c (ha-logbook.ts:471:1)
    at w (ha-logbook.ts:471:1)
    at r.value (dfe632d8.js:1:10011)
    at r.value (dfe632d8.js:1:9536)
    at r.performUpdate (styles.ts:372:1)
    at r.scheduleUpdate (styles.ts:372:1)
```

**Comment by TimoJJ** [comment]
Problem was because several Mariadb table values had changed to 'none' during the update to HA 2023.3 beta. Restored full system from backup and will check if this happens again with update to 2023.3 release version.

**Comment by davmc123** [comment]
Did this occur in 2023.3 for you? 

I'm having this issue in 2023.3.6 (although it may have started in 2023.3). All entities apart from recently created ones display an 'Unknown error' in statistics cards.

 Is there any way to repair the database?

**Comment by TimoJJ** [comment]
It happened also with 2023.3 release. I think there were several issues with MariaDB conversion or recorder service that were fixed in the latest builds. 
After few unsuccessful updates to 2023.3.1 and .2 builds I removed MariaDB addon, reinstalled it and started with a new database. 
I think there are ways to repair it, read HA community's release thread "2023.3: Dialogs!". I didn't bother since I had no critical history data.

**Comment by davmc123** [comment]
Thanks the response. I missed that you were using a MariaDB database rather than the default SQLite one. My error messages therefore look different, but still seems to be related to type issues, so could have the same underlying cause.

Deleting the database clears the error, but isn't optimal for my main install as it deletes my historical energy data. 

I will therefore need to figure out how I can repair it or hope that it gets fixed in a future update.

```
2023-04-09T23:23:38.915054939Z 2023-04-10 00:23:38.913 ERROR (DbWorker_2) [homeassistant.components.recorder.util] Error executing query: 'NoneType' object cannot be interpreted as an integer
2023-04-09T23:23:38.967879393Z Traceback (most recent call last):
2023-04-09T23:23:38.968028633Z   File "/usr/src/homeassistant/homeassistant/components/recorder/util.py", line 129, in session_scope
2023-04-09T23:23:38.968083158Z     yield session
2023-04-09T23:23:38.968115678Z   File "/usr/src/homeassistant/homeassistant/components/recorder/statistics.py", line 1380, in statistic_during_period
2023-04-09T23:23:38.968156941Z     oldest_5_min_stat = _first_statistic(
2023-04-09T23:23:38.968188020Z   File "/usr/src/homeassistant/homeassistant/components/recorder/statistics.py", line 1201, in _first_statistic
2023-04-09T23:23:38.968227669Z     return dt_util.utc_from_timestamp(stats[0].start_ts)
2023-04-09T23:23:38.968262875Z TypeError: 'NoneType' object cannot be interpreted as an integer
2023-04-09T23:23:38.968311625Z 2023-04-10 00:23:38.915 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [139963493500528] Error handling message: Unknown error (unknown_error) "REDACTED_USER" from "REDACTED_IP_ADDRESS" (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36)
2023-04-09T23:23:38.968380521Z Traceback (most recent call last):
2023-04-09T23:23:38.968410042Z   File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 26, in _handle_async_response
2023-04-09T23:23:38.968445096Z     await func(hass, connection, msg)
2023-04-09T23:23:38.968476218Z   File "/usr/src/homeassistant/homeassistant/components/recorder/websocket_api.py", line 137, in ws_get_statistic_during_period
2023-04-09T23:23:38.968513154Z     await get_instance(hass).async_add_executor_job(
2023-04-09T23:23:38.968542452Z   File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
2023-04-09T23:23:38.968578439Z     result = self.fn(*self.args, **self.kwargs)
2023-04-09T23:23:38.968611183Z   File "/usr/src/homeassistant/homeassistant/components/recorder/websocket_api.py", line 106, in _ws_get_statistic_during_period
2023-04-09T23:23:38.968650801Z     statistic_during_period(
2023-04-09T23:23:38.968677215Z   File "/usr/src/homeassistant/homeassistant/components/recorder/statistics.py", line 1380, in statistic_during_period
2023-04-09T23:23:38.968709847Z     oldest_5_min_stat = _first_statistic(
2023-04-09T23:23:38.968735954Z   File "/usr/src/homeassistant/homeassistant/components/recorder/statistics.py", line 1201, in _first_statistic
2023-04-09T23:23:38.968769242Z     return dt_util.utc_from_timestamp(stats[0].start_ts)
2023-04-09T23:23:38.968797016Z TypeError: 'NoneType' object cannot be interpreted as an integer
```

**Comment by atmezferix** [comment]
I've also got the error with the Statistics and the unknown error.  I am also running MariaDb and my Stats and Energy both broke as part of the 2023.3 upgrade.  Thankfully the recent 2023.4 upgrade has fixed my Energy history now and was hoping it would resolve my statistics but no such luck :-( Do you think its possible to fix this? And/or will there be further update that might resolve it ?  Happy to share any errors / data or logs if you need them to troubleshoot further.

**Comment by davmc123** [comment]
I ended up managing to sort it by opening the SQLite database with the tool from https://sqlitebrowser.org/ and deleting some rows from the 'statistics_short_term' table. I guessed from the error that it was because a number of statistics were missing 'created_ts' and 'start_ts'. Deleting all the entries that were missing data from those fields seems to have fixed the problem. 

All the bad data says it was created on 3rd March in the 'created' column, which I assume was after when watchtower updated my home assistant image from the stable channel.

![image](https://user-images.githubusercontent.com/9118946/231576745-900afab8-e323-4378-bcd4-ecb8c4823fab.png)

You would need a different tool to edit a MariaDB database and it may not be the same issue.
